sound/soc/sof/core.c

```diff
@@ -288,35 +298,51 @@ static int sof_probe(struct platform_device *pdev)
 		goto dbg_err;
 	}
 
+	/*
+	 * let's register miscdev now.
+	 * This will be used by the device manager in hypervisor
+	 * so that vFE can communicate with vBE
+	 */
+	sof_virtio_miscdev_register(sdev);
+
 	/* init the IPC */
 	sdev->ipc = snd_sof_ipc_init(sdev);
 	if (!sdev->ipc) {
 		dev_err(sdev->dev, "error: failed to init DSP IPC %d\n", ret);
 		goto ipc_err;
 	}
 
-	/* load the firmware */
-	ret = snd_sof_load_firmware(sdev, true);
-	if (ret < 0) {
-		dev_err(sdev->dev, "error: failed to load DSP firmware %d\n",
-			ret);
-		goto fw_load_err;
-	}
+	/* vFE will not touch HW. Let's skip fw loading */
+	if (!plat_data->is_vfe) {
```

Comment:
Can we do something like 
```
if (sdev->is_vfe)
   goto next_part of_init;
```
This keeps the indentation less.

It's probably also worthwhile having a function for setting up sdev->vfe variables so they can be done together with next_comp_id (i.e. all vfe init logic is in one place)

Reply:
Yes, "goto" is I want to use at first. Thinking of too much goto may be not comfortable to reviewers, I removed "goto". I will use goto :) and set sdev->is_vfe.